src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala

```diff
@@ -7,29 +7,30 @@ import org.apache.mesos.Protos.TaskID
 /**
   * Utility functions for dealing with TaskIDs
   */
-
 class TaskIdUtil {
-  val taskDelimiter = "."
+  val appDelimiter = "."
   val TaskIdRegex = """^(.+)[\._]([^_\.]+)$""".r
   val uuidGenerator = Generators.timeBasedGenerator(EthernetAddress.fromInterface())
 
-  def taskId(appId: PathId): String = {
-    appId.safePath + taskDelimiter + uuidGenerator.generate()
-  }
-
   def appId(taskId: TaskID): PathId = appId(taskId.getValue)
 
+  // this should actually be a try or option
   def appId(taskId: String): PathId = {
     taskId match {
       case TaskIdRegex(appId, uuid) => PathId.fromSafePath(appId)
+      case _                        => throw new MatchError(s"taskId $taskId is no valid identifier")
```

Comment:
Well, yes. Actually, appId should return an option. But the regex match was at some point introduced without doing that, so MatchErrors would throw without any meaningful message, only something like `scala.MatchError:  (of class java.lang.String)` in case of an empty String. I was too lazy to change the signature to return an Option, but at least invalid taskIds are logged now :)